models/Asset.php

```diff
@@ -1108,6 +1111,21 @@ public function getStream()
         return $this->stream;
     }
 
+    public function getChecksum(): string
+    {
+        if (!$checksum = $this->getCustomSetting('checksum')) {
+            $this->generateChecksum();
+            $checksum = $this->getCustomSetting('checksum');
+        }
+
+        return $checksum;
+    }
+
+    public function generateChecksum(): void
```

Comment:
```suggestion
    private function generateChecksum(): void
```
We should have this either `private` or `@internal`.

Reply:
Marked internal